src/Components/Aspire.Azure.Messaging.WebPubSub/AspireWebPubSubExtensions.cs

```diff
@@ -74,6 +74,29 @@ public static void AddKeyedAzureWebPubSubServiceClient(
         new WebPubSubComponent().AddClient(builder, DefaultConfigSectionName, configureWithServiceKeyAsDefaultHubName, configureClientBuilder, connectionName: connectionName, serviceKey: serviceKey);
     }
 
+    /// <summary>
+    /// Registers <see cref="WebPubSubServiceClient"/> as a singleton for given <paramref name="connectionName"/> in the services provided by the <paramref name="builder"/>. This
+    /// overload does not require a service key and uses the connection name as the service key
+    /// to support scenarios where multiple Hubs are referenced in the same application.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">The name of the component to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="AzureMessagingWebPubSubSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientBuilder">An optional method that can be used for customizing the <see cref="IAzureClientBuilder{WebPubSubServiceClient, WebPubSubServiceClientOptions}"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire.Azure.Messaging.WebPubSub:{name}" section.</remarks>
+    /// <exception cref="InvalidOperationException">Thrown when neither <see cref="AzureMessagingWebPubSubSettings.ConnectionString"/> nor <see cref="AzureMessagingWebPubSubSettings.Endpoint"/> is provided.</exception>
+    public static void AddKeyedAzureWebPubSubServiceClient(
+        this IHostApplicationBuilder builder,
+        string connectionName,
```

Comment:
what if we have 2 wps services with the same hub name which sounds a valid case, e.g. in the Host app:
```
var wps1 = builder.AddAzureWebPubSub("wps1");
var hub1 = wps1.AddHub("chat");
var wps2 = builder.AddAzureWebPubSub("wps2");
var hub2 = wps2.AddHub("chat");
var web = ...AddReference(hub1).AddReference(hub2);

```

Reply:
We don't have an overload for the WebPubSub client integration that supports providing a hub name that differs from the resource name. We do this for Azure Event Hubs (see https://github.com/dotnet/aspire/blob/9480d26740a699497aab0d61260a1f739ce786a6/src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs#L139) and it allows you to have two hubs with the same hub name but different resource names to distinguish them on the Aspire side.

Reply:
shall we do the same for Web PubSub Hub now when it supports deep-link? Previously since this it is not reference-able, there is no need to allow resource name to be customizable. Besides the same hub name scenario, another concern is that currently there is an implicit restriction that if WebPubSub resource name should not be the same as the hub name otherwise it is overwritten:
```
var wpsChat = builder.AddAzureWebPubSub("chat");
var chatHub =  builder.AddAzureWebPubSub("wps1").AddHub("chat");
var web = ....AddReference(wpsChat).AddReference(chatHub); // the latter one overwrites the former one?
```

Reply:
This is done in https://github.com/dotnet/aspire/pull/8235/commits/e97054f0c25936f3b753be89c93115492c8837ca. I placed on obsoletion marker on the old API in https://github.com/dotnet/aspire/pull/8235/commits/33e797663d12cc91cfd32cb6a4ddbee6c89f9f40.